src/winetricks

```diff
@@ -1028,58 +1044,73 @@ winetricks_selfupdate_rollback()
 # Caches downloads in winetrickscache/$packagename
 w_download_to()
 {
+		# Export downloader and/or torify
     winetricks_download_setup
 
     _W_packagename="$1" # or path to download file to
-    _W_url="$2"
-    _W_sum="$3"
-    _W_file="$4"
-    _W_cookiejar="$5"
+    _W_url="$2" # Any url
+    _W_sum="$3" # Any shasum hash
+    _W_file="$4" # Any file
+    _W_cookiejar="$5" # QA: wtf?
 
+		# Sanity _W_packagename
     case $_W_packagename in
-        .) w_die "bug: please do not download packages to top of cache" ;;
+				# QA required: What is top of cache?
+        .) w_die "First argument '$_W_packagename' in 'w_download_to' function in invalid - Do not download packages to top of cache" ;;
+				"") w_die "First argument '$_W_packagename' in 'w_download_to' function is blank which is unexpected" ;;
+				[!/]*) # Pattern for absolute unix path
+						w_debug "First variable '$_W_packagename' in 'w_download_to' function is valid absolute unix path"
+						# Export path
+						if ! grep -q '^/' "$_W_packagename"; then
+		            _W_cache="$W_CACHE/$_W_packagename"
+		        else
+		            _W_cache="$_W_packagename"
+		        fi
+				;;
+				*) w_debug "First variable '$_W_packagename' in 'w_download_to' function is valid, using file name"
```

Comment:
Formatting is completely foobarred (here and other places).

Reply:
which is matched above in case it's not following:
```sh
*[0-9a-zA-z/_.:%]*) _W_url="${_W_url// /\%20}" ;;
			*[0-9a-zA-z/_.:]*) w_debug "Second variable '$_W_url' in 'w_download_to' function is valid" ;;
			*) w_die "Second argument '$_W_url' in 'w_download_to' variable is blank which is unexpected"
```
in case unicode or illegal characters are used, or what do you suggest?